src/lib/storage/fixedstring_dictionary_column/fixedstring_encoder.hpp

```diff
@@ -0,0 +1,117 @@
+#pragma once
+
+#include <algorithm>
+#include <limits>
+#include <memory>
+
+#include "storage/base_column_encoder.hpp"
+
+#include "storage/dictionary_column.hpp"
+#include "storage/fixedstring_dictionary_column/fixedstring_column.hpp"
+#include "storage/value_column.hpp"
+#include "storage/vector_compression/base_compressed_vector.hpp"
+#include "storage/vector_compression/vector_compression.hpp"
+#include "types.hpp"
+#include "utils/enum_constant.hpp"
+
+namespace opossum {
+
+/**
+ * @brief Encodes a column using dictionary encoding and compresses its attribute vector using vector compression.
+ *
+ * The algorithm first creates an attribute vector of standard size (uint32_t) and then compresses it
+ * using fixed-size byte-aligned encoding.
+ */
+class FixedStringDictionaryEncoder : public ColumnEncoder<FixedStringDictionaryEncoder> {
```

Comment:
This also looks like something were we can avoid duplication by adding a template parameter